components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java

```diff
@@ -322,64 +322,34 @@ public AuthenticationContext authenticateForWebSocket(SignedJWT jwtToken, String
-
-                    if (log.isDebugEnabled()) {
-                        log.debug("Subscription validation via Key Manager. Status: " +
-                                apiKeyValidationInfoDTO.isAuthorized());
-                    }
-                    if (apiKeyValidationInfoDTO.isAuthorized()) {
-                        log.debug("JWT authentication successful.");
-                        String endUserToken;
-                        if (jwtGenerationEnabled) {
-                            JWTInfoDto jwtInfoDto;
-                            try {
-                                jwtInfoDto =
-                                        GatewayUtils.generateJWTInfoDto(jwtValidationInfo, api, apiKeyValidationInfoDTO,
-                                                apiContext, apiVersion);
-                                endUserToken = generateAndRetrieveJWTToken(tokenSignature, jwtInfoDto);
-                                return GatewayUtils
-                                        .generateAuthenticationContext(tokenSignature, jwtValidationInfo, null,
-                                                apiKeyValidationInfoDTO, getApiLevelPolicy(), endUserToken, true);
-                            } catch (ParseException e) {
-                                throw new APISecurityException(APISecurityConstants.API_AUTH_GENERAL_ERROR,
-                                        APISecurityConstants.API_AUTH_GENERAL_ERROR_MESSAGE);
-                            }
-                        }
-                    } else {
-                        log.debug("User is NOT authorized to access the Resource. API Subscription validation failed.");
-                        throw new APISecurityException(apiKeyValidationInfoDTO.getValidationStatus(),
-                                "User is NOT authorized to access the Resource. API Subscription validation failed.");
-                    }
-                }
-                log.debug("Ignored subscription validation");
+            log.debug("Begin subscription validation via Key Manager");
+            APIKeyValidationInfoDTO apiKeyValidationInfoDTO = validateSubscriptionUsingKeyManager(apiContext,
+                    apiVersion, jwtValidationInfo);
+
+            if (log.isDebugEnabled()) {
+                log.debug("Subscription validation via Key Manager. Status: " +
+                        apiKeyValidationInfoDTO.isAuthorized());
             }
-            log.debug("JWT authentication successful.");
-            String endUserToken = null;
-            try {
-                if (jwtGenerationEnabled) {
-                    JWTInfoDto jwtInfoDto =
-                            GatewayUtils.generateJWTInfoDto(jwtValidationInfo, api, null, apiContext, apiVersion);
-                    endUserToken = generateAndRetrieveJWTToken(tokenSignature, jwtInfoDto);
+            if (apiKeyValidationInfoDTO.isAuthorized()) {
+                log.debug("JWT authentication successful.");
```

Comment:
can we append the username/any other info here?

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/8999/commits/83d8083bf4fdd8144025e55cdf9c2c0259d88850